packages/web-components/src/calendar/calendar.spec.md

```diff
@@ -0,0 +1,126 @@
+
+### Calendar vs. Date Picker Implementation
+
+The calendar and the date picker have significant overlap in inputs, outputs, and events. The difference is that the date picker is essentially a calendar combined with an input field, in which the input field can can interact with the calendar by opening and closing the calendar and displaying/modifying the selected dates of the calendar. It is important to note that the calendar is the component that stores the selected dates, which is then exposed to the date picker.
+
+## Design Spec
+
+Due to the incomplete nature of the v9 calendar design spec, work on the component will be based on the calendar from the v8 api while using the skin of the calendar in the v9 date picker design spec.
+
+[Link to v9 Date Picker Design in Figma](https://www.figma.com/file/9EHmS5y1Rr7DCh7KgJBwuz/DatePicker?type=design&mode=design&t=E6sL2ffuuZfwCSRp-0)
+
+[Link to Fluent v8 Calendar Component](https://developer.microsoft.com/en-us/fluentui#/controls/web/calendar)
+
+## Engineering Spec
+
+### Inputs
+
+| attribute                  | type                                                    | default           | description                                                                                                                       |
+| -------------------------- | ------------------------------------------------------- | ----------------- | --------------------------------------------------------------------------------------------------------------------------------- |
+| `type`                     | "date" \| "month" \| "week" \| "year" \| "range-picker" | "date"            | The type of the calendar                                                                                                          |
+| `month-picker-visible`     | boolean                                                 | `true`            | Whether the month picker is shown beside the day picker or hidden                                                                 |
+| `month-picker-overlay`     | boolean                                                 | -                 | If true, show month picker on top of date picker when visible                                                                     |
+| `week-numbers`             | boolean                                                 | -                 | If true, show week numbers (1-53) before each week row                                                                            |
+| `min-weeks`                | number                                                  | -                 | Minimum number of weeks shown in a month                                                                                          |
+| `filter`                   | "week" \| "work-week" \| "4" \| "3" \| "2" \| "1"       | -                 | Allow user to view selected date with a range filter                                                                              |
```

Comment:
Is there a more idiomatic way of expressing this filter union type? I don't like the mixture of the numbers and the numbers as strings. It's not obvious what the numbers are doing.

Reply:
I 100% agree! The numbers represent number of days to filter by (it's spec-ed in the "variation of selection based on filter" section in the [DatePicker – Figma](https://www.figma.com/file/9EHmS5y1Rr7DCh7KgJBwuz/DatePicker?type=design&node-id=133-15&mode=design&t=G8aEWuokJnFl5R1T-0)) I definitely want to keep the "week" and "work-week" options since the idea is that they can vary by locale. What do you think about changing the numbers in to "four," "three" etc.

Reply:
> Is there a more idiomatic way of expressing this filter union type? I don't like the mixture of the numbers and the numbers as strings. It's not obvious what the numbers are doing.

I 100% agree! The numbers represent number of days to filter by (it's spec-ed in the "variation of selection based on filter" section in the [DatePicker – Figma](https://www.figma.com/file/9EHmS5y1Rr7DCh7KgJBwuz/DatePicker?type=design&node-id=133-15&mode=design&t=G8aEWuokJnFl5R1T-0)) I definitely want to keep the "week" and "work-week" options since the idea is that they can vary by locale. What do you think about changing the numbers in to "four," "three" etc.